public/app/plugins/datasource/prometheus/querybuilder/shared/OperationHeader.tsx

```diff
@@ -79,7 +79,14 @@ export const OperationHeader = React.memo<Props>(
                 if (value.value) {
                   // Operation should exist if it is selectable
                   const newDef = queryModeller.getOperationDef(value.value.id)!;
-                  let changedOp = { ...operation, id: value.value.id };
+
+                  // copy default params, and override with all current params
+                  const newParams = [...newDef.defaultParams];
+                  for (let i = 0; i < Math.min(operation.params.length, newParams.length); i++) {
+                    newParams[i] = operation.params[i];
```

Comment:
should we check param names or something? Or expected value? What will happen if in one op param is label but in second it should be a number value?

Reply:
We should definitely improve the handling here. I.e. changing a operation to `Binary operation with query` does currently nothing.

I would go with the following:
1. improve general "cloning" of operations in the `onChange` handler in this PR
2. handle edge cases like using the operations `addOperationHandler` in the fix for #51780

Is that fine for you @aocenas ?

Reply:
Yeah sure, honestly maybe just checking type is ok but will defer that to you just wanted to ask if that was considered and we don't end up with having a label name in a field expecting number which I assume could also blow up somewhere.